src/posit/connect/tasks.py

```diff
@@ -0,0 +1,204 @@
+        """
+        return self["output"]
+
+    @property
+    def error_code(self) -> int | None:
+        """The error code.
+
+        The error code produced by the task. A non-zero value represent an
+        error. A zero value represents no error.
+
+        Returns
+        -------
+        int | None
+            Non-zero value indicates an error.
+        """
+        return self["code"] if self.is_finished else None
+
+    @property
+    def error_message(self) -> str | None:
+        """The error message.
+
+        Returns
+        -------
+        str | None
+            Human readable error message, or None on success or not finished.
+        """
+        return self.get("error") if self.is_finished else None
+
+    @property
+    def result(self) -> dict | None:
+        """The task result.
+
+        Returns
+        -------
+        dict | None
+        """
+        return self.get("result")
+
+    # CRUD Methods
+
+    @overload
+    def update(self, first: int, wait: int, **kwargs) -> None:
+        """Update the task.
+
+        Parameters
+        ----------
+        first : int, default 0
+            Line to start output on.
+        wait : int, default 0
+            Maximum number of seconds to wait for the task to complete.
+        """
+        ...
+
+    @overload
+    def update(self, *args, **kwargs) -> None:
+        """Update the task."""
+        ...
+
+    def update(self, *args, **kwargs) -> None:
+        """Update the task.
+
+        See Also
+        --------
+        task.wait_for : Wait for the task to complete.
+
+        Notes
+        -----
+        When waiting for a task to complete, one should consider utilizing `task.wait_for`.
+
+        Examples
+        --------
+        >>> task.output
+        [
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
+        ]
+        >>> task.update()
+        >>> task.output
+        [
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
+            "Pretium aenean pharetra magna ac placerat vestibulum lectus mauris."
+        ]
+        """
+        params = dict(*args, **kwargs)
+        path = f"v1/tasks/{self.id}"
+        url = urls.append(self.config.url, path)
+        response = self.session.get(url, params=params)
+        result = response.json()
+        super().update(**result)
+
+    def wait_for(self, sleep: int = 1) -> None:
+        """Wait for the task to finish.
+
+        Parameters
+        ----------
+        sleep : int, optional
+            Maximum number of seconds to wait between status checks.
+
+        Examples
+        --------
+        >>> task.wait_for()
+        None
+
+        Wait five seconds between status checks.
+        >>> task.wait_for(5)
+        None
+        """
+        while not self.is_finished:
+            self.update(wait=sleep)
```

Comment:
I know the API supports this, but it's odd to me that we'd push the `sleep` to the server. I.e. with `wait_for(5)`, each request is going to take 5 seconds to respond. Seems odd to consume server resources for that waiting. Why not do the sleeping on the client side and always send wait=0 (which is the default anyway)?

The other thing you might do is pass the `first` argument, set to `len(self.output)` so that you're only returning newly generated lines of log output. You'd have to do something different than just `super().update(**result)` with that, have to be sure to append output and update the rest. But I could see that for a large task (IDK, lots of package installation output), not fetching all lines every time could be important.

Reply:
I had the same reaction when developing this. The one advantage of the current implementation is that the API will return once the task is finished. So, if you call with `wait=60`, the method will return in less than 60 seconds if the task finishes. 

But I also agree. It seems like an anti-pattern to consume server threads waiting for the task to complete.

Reply:
Added #184